In [ ]:
1#Read the dataset to python environment.
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
student_data=pd.read_csv('/content/StudentsPerformance.csv')
student_data

gender race/ethnicity  ... reading score writing score
0    female        group C  ...            17            10
1    female        group B  ...            24            23
2    female        group B  ...            32            28
3    female        group B  ...            38            32
4    female        group C  ...            39            33
..      ...            ...  ...           ...           ...
995    male        group E  ...            87            81
996    male        group A  ...            96            86
997    male        group D  ...            97            99
998    male        group E  ...           100            93
999    male        group E  ...           100           100

[1000 rows x 8 columns]

In [ ]:
#1Find out how many males and females participated in the test.
student_data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

**518 female and 482 male participated in test**

In [ ]:
#What do you think about the students' parental level of education?
student_data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Out of the total 1000 parents,226 went to some colleges, 228 have associate degree, 196 has high school education, 179 went to some high school, 118 has bachelors degree and 59 has masters degree.This indicate that there are no illetrate parents.

In [ ]:
#3.a Who scores the most on average for math, reading and writing based on
● Gender



In [ ]:
student_data2=student_data[student_data["gender"] == "female"]  
student_data2                  

gender race/ethnicity  ... reading score writing score
0    female        group C  ...            17            10
1    female        group B  ...            24            23
2    female        group B  ...            32            28
3    female        group B  ...            38            32
4    female        group C  ...            39            33
..      ...            ...  ...           ...           ...
513  female        group E  ...           100           100
514  female        group E  ...            93            90
515  female        group E  ...            92            97
516  female        group E  ...           100           100
517  female        group E  ...           100           100

[518 rows x 8 columns]

In [ ]:
student_data2.columns


Index(['gender', 'race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course', 'math score', 'reading score',
       'writing score'],
      dtype='object')

In [ ]:
student_data2.drop(['race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course'],axis=1)

gender  math score  reading score  writing score
0    female           0             17             10
1    female           8             24             23
2    female          18             32             28
3    female          11             38             32
4    female          22             39             33
..      ...         ...            ...            ...
513  female          99            100            100
514  female          99             93             90
515  female         100             92             97
516  female         100            100            100
517  female         100            100            100

[518 rows x 4 columns]

In [ ]:
student_data2['math score'].mean()

63.37644787644788

In [ ]:
student_data2['reading score'].mean()

72.59073359073359

In [ ]:
student_data2['writing score'].mean()

72.46718146718146

In [ ]:
student_data3=student_data[student_data["gender"] == "male"]  
student_data3  

gender race/ethnicity  ... reading score writing score
518   male        group C  ...            34            36
519   male        group A  ...            23            19
520   male        group B  ...            24            15
521   male        group E  ...            26            22
522   male        group B  ...            32            34
..     ...            ...  ...           ...           ...
995   male        group E  ...            87            81
996   male        group A  ...            96            86
997   male        group D  ...            97            99
998   male        group E  ...           100            93
999   male        group E  ...           100           100

[482 rows x 8 columns]

In [ ]:
student_data3.drop(['race/ethnicity', 'parental level of education', 'lunch',
       'test preparation course'],axis=1)

gender  math score  reading score  writing score
518   male          27             34             36
519   male          28             23             19
520   male          30             24             15
521   male          30             26             22
522   male          31             32             34
..     ...         ...            ...            ...
995   male          99             87             81
996   male         100             96             86
997   male         100             97             99
998   male         100            100             93
999   male         100            100            100

[482 rows x 4 columns]

In [ ]:
student_data3['math score'].mean()

68.82157676348548

In [ ]:
student_data3['reading score'].mean()

65.54564315352697

In [ ]:
student_data3['writing score'].mean()

63.446058091286304

**Conclusion:**
1. Males scores most on average for maths
2. Female scores most on average for reading test
3. Female scores most on average for writing test


In [ ]:
#3.b Who scores the most on average for math, reading and writing based on
● Tst preparation course

In [ ]:
student_data4=student_data['test preparation course'].tolist()
student_data4

['none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'completed',
 'completed',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',
 'completed',
 'none',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'completed',
 'completed',
 'none',
 'none',
 'none',
 'none',
 'none',
 'completed',
 'none',


In [ ]:
student_data4=student_data[student_data["test preparation course"] == "none"]  
student_data4  

gender race/ethnicity  ... reading score writing score
0    female        group C  ...            17            10
1    female        group B  ...            24            23
2    female        group B  ...            32            28
3    female        group B  ...            38            32
4    female        group C  ...            39            33
..      ...            ...  ...           ...           ...
984    male        group E  ...            88            78
986    male        group E  ...            73            71
987    male        group D  ...            81            84
990    male        group C  ...            93            91
991    male        group E  ...            87            82

[642 rows x 8 columns]

In [ ]:
student_data4['math score'].mean()

63.78971962616822

In [ ]:
student_data4['reading score'].mean()

66.41744548286604

In [ ]:
student_data4['writing score'].mean()

64.45794392523365

In [ ]:
student_data5=student_data[student_data["test preparation course"] == "completed"]  
student_data5

gender race/ethnicity  ... reading score writing score
5    female        group B  ...            44            36
10   female        group C  ...            40            44
12   female        group B  ...            51            44
16   female        group A  ...            48            41
21   female        group D  ...            55            60
..      ...            ...  ...           ...           ...
995    male        group E  ...            87            81
996    male        group A  ...            96            86
997    male        group D  ...            97            99
998    male        group E  ...           100            93
999    male        group E  ...           100           100

[358 rows x 8 columns]

In [ ]:
student_data5['math score'].mean()

69.96648044692738

In [ ]:
student_data5['reading score'].mean()

74.17597765363128

In [ ]:
student_data5['writing score'].mean()

74.68435754189944

**Conclusion**
1.Students who completed the preaparatory course has greater average in maths, reading and writing comparaed to the students who didnt attend any preparatory course.

In [ ]:
#What do you think about the scoring variation for math, reading and writing
based on
*Gender


In [ ]:
import statistics

In [ ]:
student_data6=student_data2.var()
student_data6


math score       256.958593
reading score    207.677438
writing score    220.369327
dtype: float64

In [ ]:
student_data7=student_data3.var()
student_data7

math score       211.889097
reading score    200.211010
writing score    202.413924
dtype: float64

**Conclusion#**
Variance in math score, reading and writing test is greater for female compared to males.


In [ ]:
student_data8=student_data4.var()
student_data8

math score       246.668662
reading score    213.419851
writing score    226.251739
dtype: float64

In [ ]:
student_data9=student_data5.var()
student_data9

math score       210.884027
reading score    183.265864
writing score    175.202612
dtype: float64

**Conclusion# **
Variance in math score, reading and writing test is greater for students who didnt attend any preparatory course when compared to students who attended the preparatory course.

In [ ]:
#The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this

In [ ]:
student_data10=student_data.drop(['parental level of education', 'lunch',
       'test preparation course', 'reading score',
       'writing score'],axis=1)
student_data10
      

gender race/ethnicity  math score
0    female        group C           0
1    female        group B           8
2    female        group B          18
3    female        group B          11
4    female        group C          22
..      ...            ...         ...
995    male        group E          99
996    male        group A         100
997    male        group D         100
998    male        group E         100
999    male        group E         100

[1000 rows x 3 columns]

In [ ]:
student_data10.sort_values(by='math score',ascending=False)

gender race/ethnicity  math score
999    male        group E         100
996    male        group A         100
515  female        group E         100
517  female        group E         100
516  female        group E         100
..      ...            ...         ...
2    female        group B          18
51   female        group C          13
3    female        group B          11
1    female        group B           8
0    female        group C           0

[1000 rows x 3 columns]

In [ ]:
student_data10['math score'].describe()

count    1000.000000
mean       66.001000
std        15.569567
min         0.000000
25%        57.000000
50%        66.000000
75%        77.000000
max       100.000000
Name: math score, dtype: float64

There are 1000 students. As per the requirement we need to find the top 25 percentage of students according to the math score. ie top 250 students

In [ ]:
percent=student_data10.nlargest(250,'math score')
percent

gender race/ethnicity  math score
515  female        group E         100
516  female        group E         100
517  female        group E         100
996    male        group A         100
997    male        group D         100
..      ...            ...         ...
428  female        group D          77
429  female        group E          77
849    male        group A          77
850    male        group C          77
851    male        group D          77

[250 rows x 3 columns]

The above list shows top 250 students based on mark scored (given data set has 1000 student data, 25 percentage of it is 250). But when we look at the data set, there are 5 more students with score of 77 marks. So when awarding bonus, ideally those students should also be awarded. So ideally management should provide bonus to 260 students, which is greater than 25 percentage.